tripleawarclub.org/public_html/dice/dice.class.php

```diff
@@ -0,0 +1,287 @@
+<?php
+class dice {
+	var $domain;
+	var $dbconn;
+	var $db = null;
+	var $enc = [];
+
+	// constructor
+    function __construct() {
+		$this->domain = self::getBaseUri();
+		$this->connectDatabase();
+    }
+
+	// destructor
+	function __destruct() {
+		if(! is_null($this->db)){
+			$this->disconnectDatabase();
+		}
+	}
+
+	/**
+	 * Returns the base URI of the active request.  The returned URI WILL NOT have a trailing slash.
+	 */
+	static function getBaseUri() {
+		$path = $_SERVER['DOCUMENT_URI'];
+		$pathLastSegment = "/" . basename($path);
+		$pathWithoutLastSegment = substr($path, 0, -strlen($pathLastSegment));
+		return $_SERVER['REQUEST_SCHEME'] . "://" . $_SERVER['HTTP_HOST'] . $pathWithoutLastSegment;
+	}
+
+	////////////////////////////////
+	//        database            //
+	////////////////////////////////
+    function connectDatabase() {
+    	if(! is_null($this->db)){
+			return;
+		}
+
+    	$host = getenv("MARTI_DB_HOST");
+    	$user = getenv("MARTI_DB_USERNAME");
+    	$password = getenv("MARTI_DB_PASSWORD");
+    	$database = getenv("MARTI_DB_NAME");
+    	$this->dbconn = new mysqli($host, $user, $password, $database);
+    	if ($this->dbconn->connect_errno > 0) {
+    		exit("fatal error: could not connect to database!<br>" . mysqli_connect_error() . "!");
+    	}
+    	$this->db = mysqli_select_db($this->dbconn, $database);
+    }
+
+	function updateStats($numdice) {
+		$this->connectDatabase();
+
+		$sql = "UPDATE stats SET requests=requests+1, dice_rolled=dice_rolled+$numdice";
+		$result = $this->dbconn->query($sql) or exit("fatal error: data connection lost @updateStats!");
+
+	}
+
+	function getStats() {
+		$this->connectDatabase();
+
+		$sql = "SELECT * FROM stats";
+		$result = $this->dbconn->query($sql) or exit("fatal error: data connection lost @getStats!");
+		$stats = mysqli_fetch_array($result);
+		return $stats;
+	}
+
+	/**
+	 * Checks if all receiving email adresses are registered
+	 * @returns bool
+	 */
+	function checkIfMailsAreRegistered(array $emails) {
```

Comment:
Future Note: I'd argue that `checkIfMailsAreRegistered` does the same as `checkIfMailIsRegistered`...